### PR TITLE
Added option for using PyTorch for throughput testing, issue 219

### PR DESCRIPTION
**Pull request by gregw18**
Added option for using PyTorch for throughput testing, using petastom.pytorch.DataLoader.
Had to modify compat.py to return column.data.num_chunks regardless of pyarrow version, as 015 was crashing on column.num_chunks.
Added option to pass min_after_retrieve to petastorm.pytorch.DataLoader, to give it similar functionality to TensorFlow.
Note that I am just learning Python, so any and all feedback is appreciated!

### CODE REVIEW

File: petastorm/pytorch.py [hunk 1]
```diff
@@ -129,7 +129,8 @@ def __init__(self, reader, batch_size=1, collate_fn=decimal_friendly_collate,
         if shuffling_queue_capacity > 0:
             # We can not know what is the reasonable number to use for the extra capacity, so we set a huge number
             # and give up on the unbound growth protection mechanism.
-            min_after_dequeue = shuffling_queue_capacity - 1
+            if min_after_dequeue == 0:
```
[the remark this answers is not in the document]
Comment (gregw18) [reply petastorm/pytorch.py]: I'm afraid that I'm not clear what you mean. Should I just default min_after_dequeue to 0 like tf_tensors()?
I'm probably brainwashed from working with .net for so long, but I am usually hesitant to change interfaces and thus require changes from callers. My goal was to allow callers to provide the parameter if they want to, but otherwise maintain the current behavior. Would using a default value of -1 to recognize when to use the originally hard-coded value of (shuffling_queue_capacity - 1) be reasonable?

[the remark this answers is not in the document]
Comment (selitvin) [reply petastorm/pytorch.py]: Thinking about it more, you are probably right. We can keep the default as `shuffling_queue_capacity-1`.

> Would using a default value of -1 to recognize when to use the originally hard-coded value of (shuffling_queue_capacity - 1) be reasonable

How about `min_after_dequeue=None`? What I mean is that `0` in `min_after_dequeue=0` can actually make some limited sense as in `RandomShufflingBuffer(capacity, min_after_retrieve=0)` - we deplete the entire buffer before re-accumulating and reshuffling it.

**Comment by selitvin** [comment]
Do we want to land this PR? If so, can you please rebase since some changes in this PR has landed already.

File: petastorm/pytorch.py [hunk 2]
```diff
@@ -133,7 +135,8 @@ def __init__(self, reader, batch_size=1, collate_fn=decimal_friendly_collate,
         if shuffling_queue_capacity > 0:
             # We can not know what is the reasonable number to use for the extra capacity, so we set a huge number
             # and give up on the unbound growth protection mechanism.
-            min_after_dequeue = shuffling_queue_capacity - 1
+            if not min_after_dequeue:
```
Comment (selitvin) [inline petastorm/pytorch.py]: I think, if `min_after_dequeue` = 0, we would override the value, which is not the intent.

**Comment by gregw18** [comment]
Whoops - thanks for catching that!
I seem to be having some problems with the build, and they don't seem to be related to my code. I'm going to trigger another build Saturday morning to see if it will succeed then.
